### PR TITLE
Add JSON output option for fluxctl's list-images and list-workloads

### PR DESCRIPTION
**Pull request by trevrosen**
## What
A JSON output option for `fluxctl list-workloads` and `fluxctl list-images`

## Why
Building tooling (such as chatbot functionality) is much easier with a machine-readable output.

## Note
I added [Testify](https://github.com/stretchr/testify) to the test build b/c I think that the single-line assertions make test code really easy to read. If this is controversial, I can remove. It tends to be the single thing that I use outside of stdlib for tests.

### CODE REVIEW

File: cmd/fluxctl/format.go [hunk 1]
```diff
@@ -29,3 +45,142 @@ func makeExample(examples ...string) string {
...
+		if f == format {
+			return true
+		}
+	}
+	return false
+}
+
+// outputImagesJson sends the provided ImageStatus info to the io.Writer in JSON formatting, honoring limits in opts
+func outputImagesJson(images []v6.ImageStatus, out io.Writer, opts *imageListOpts) error {
+	if opts.limit < 1 {
```
Comment (hiddeco) [inline cmd/fluxctl/format.go]: Making `opts.limit == 0` equal to an infinite amount of results would be an better option in my opinion. This also makes it easier to programmatically make use of the CLI flag when you sometimes do want to limit the result.

Comment (trevrosen) [reply cmd/fluxctl/format.go]: Agreed - also that's what the usage note says anyway, so... my bad 😸 

Fixed.

Comment (trevrosen) [reply cmd/fluxctl/format.go]: ahh actually one sec need to do a bit more

Comment (hiddeco) [reply cmd/fluxctl/format.go]: In addition I would make the returned error a bit more humane to not step on anyone's toes, while also informing them about the check that failed.

My suggestion would be something like the following: 

> opts.limit is less than 0 but only supports non-negative integers

Comment (trevrosen) [reply cmd/fluxctl/format.go]: Done!

**Pull request merged**